packages/react-dom/src/events/ReactBrowserEventEmitter.js

```diff
@@ -81,6 +81,15 @@ let reactTopListenersCounter = 0;
  */
 const topListenersIDKey = '_reactListenersID' + ('' + Math.random()).slice(2);
 
+function getEventsFromRegistraionName(registrationName) {
+  if (!(registrationName in registrationNameDependencies)) {
+    registrationNameDependencies[registrationName] = [
+      `top${registrationName.slice(2)}`,
```

Comment:
Purely for discussion but...

It's really a bummer we need top level event name constants period. There's nothing special about them. We could probably just use the event name that comes from the DOM.

If we dropped top level event name constants, we could avoid binding all events, eliminating overhead for local listeners:
https://github.com/facebook/react/blob/master/packages/react-dom/src/events/ReactDOMEventListener.js#L126

Maybe that's too aggressive. 🤷‍♂️

Reply:
Yeah I was thinking the same thing. There doesn't seem to be much value to them they are just the event name with 'top' appended.

Reply:
It depends on how it affects bundle size. If we keep them and do something like https://github.com/facebook/react/pull/11894 then the event plugins can just use the numeric constants for dependencies/switching on events, which would probably compress better.